dashboards-observability/public/components/application_analytics/home.tsx

```diff
@@ -238,8 +276,94 @@ export const Home = (props: HomeProps) => {
       });
   };
 
+  // Add visualization to application's panel
+  const addVisualizationToPanel = async (
+    appId: string,
+    visualizationId: string,
+    panelId: string
+  ) => {
+    return http
+      .post(`${CUSTOM_PANELS_API_PREFIX}/visualizations`, {
+        body: JSON.stringify({
+          panelId,
+          savedVisualizationId: visualizationId,
+        }),
+      })
+      .then(() => {
+        fetchAppById(
+          http,
+          pplService,
+          appId,
+          setApplication,
+          setAppConfigs,
+          setVisWithAvailability,
+          setToasts
+        );
+      })
+      .catch((err) => {
+        // setToasts(`Error in adding ${visualizationName} visualization to the panel`, 'danger');
+        console.error(err);
+      });
+  };
+  const setStartTimeForApp = (appName: string, newStartTime: string) => {
```

Comment:
Is it really required to set time in the session? If you have copied the code from somewhere and using the same functions, then wouldn't it overwrite the session variables? Can't we save the data in Redux? what is the use of this data?